File: skills/clawhub/wecom-agent-ops-center/test-agent.js
/**
 * 测试 Agent - 模拟用户 Agent 端点，用于 E2E 测试
 *
 * 启动：node test-agent.js
 * 监听：http://localhost:3000/chat
 *
 * 收到企微消息后，原样回显（加上前缀）
 */

const http = require('http');

const PORT = process.env.TEST_AGENT_PORT || 3000;

const server = http.createServer(async (req, res) => {
  // CORS
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'POST, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    res.writeHead(204);
    res.end();
    return;
  }

  if (req.method !== 'POST' || req.url !== '/chat') {
    res.writeHead(404);
    res.end('Not Found');
    return;
  }

  // 读取请求体
  let body = '';
  req.on('data', chunk => body += chunk);

  req.on('end', () => {
    try {
      const msg = JSON.parse(body);

      const userId = msg.from && msg.from.user_id || '?';
      const content = msg.content || '';
      const msgId = msg.msg_id || '?';

      console.log(`\n📩 收到消息 | from=${userId} | "${content.slice(0, 50)}"`);

      // 回显消息（模拟 Agent 回复）
      const reply = {
        reply_to: msgId,
        content: `🤖 [测试Agent] 收到你的消息：「${content}」\n\n> 来自：${userId}\n> 时间：${new Date().toLocaleString('zh-CN')}`,
        msg_type: 'markdown',
      };

      res.writeHead(200, { 'Content-Type': 'application/json' });
      res.end(JSON.stringify(reply));

      console.log(`📤 回复 | "${reply.content.slice(0, 100)}..."`);
    } catch (e) {
      console.error(`❌ JSON 解析失败: ${e.message}`);
      res.writeHead(400);
      res.end(JSON.stringify({ error: 'Invalid JSON' }));
    }
  });
});

server.listen(PORT, () => {
  console.log(`🧪 测试 Agent 已启动: http://localhost:${PORT}/chat`);
  console.log('   等待 Connector 转发企微消息...\n');
});

File: skills/clawhub/wecom-agent-ops-center/test_agent.py
"""
模拟 Agent — 用于端到端测试 Connector

启动: python test_agent.py
监听: http://localhost:3000/chat

收到的消息会打印到控制台，自动返回确认消息。
"""

import json
import logging
from datetime import datetime
from aiohttp import web

logging.basicConfig(level=logging.INFO, format="%(asctime)s [Agent] %(message)s", datefmt="%H:%M:%S")
logger = logging.getLogger("test-agent")

call_count = 0


async def chat_handler(request):
    global call_count
    body = await request.json()
    call_count += 1

    msg_id = body.get("msg_id", "?")
    user_id = body.get("from", {}).get("user_id", "?")
    content = body.get("content", "")

    logger.info(f"← [{call_count}] from={user_id} | msg={msg_id[:8]}... | {content[:60]}")

    # 模拟处理：直接回确认
    reply = {
        "reply_to": msg_id,
        "content": f"✅ 已收到消息 #{call_count}\n\n"
                   f"你说: _{content[:100]}_\n\n"
                   f"> 这是模拟 Agent 的自动回复。\n"
                   f"> 替换为你的 Agent 端点即可获得真实 AI 回复。",
        "msg_type": "markdown",
    }

    logger.info(f"→ 回复: {reply['content'][:60]}...")
    return web.json_response(reply)


async def health_handler(request):
    return web.json_response({"status": "ok", "calls": call_count})


if __name__ == "__main__":
    app = web.Application()
    app.router.add_post("/chat", chat_handler)
    app.router.add_get("/", health_handler)

    logger.info("=" * 40)
    logger.info("模拟 Agent 启动")
    logger.info("  监听: http://localhost:3000/chat")
    logger.info("  健康: http://localhost:3000/")
    logger.info("=" * 40)

    web.run_app(app, host="0.0.0.0", port=3000)
